src/plugins/intel_cpu/tests/functional/shared_tests_instances/snippets/mha.cpp

```diff
@@ -132,6 +134,70 @@ INSTANTIATE_TEST_SUITE_P(
         ::testing::Values(enable_callback())),
     MHA::getTestCaseName);
 
+std::vector<std::vector<ov::test::InputShape>> splitm_dynamic_shapes_4d = {
+    {
+        {PartialShape{-1, -1, -1, -1}, {{1, 128, 2, 64}, {1, 17, 2, 64}, {1, 128, 2, 64}}},
+        {PartialShape{-1, -1, -1, -1}, {{1, 128, 2, 64}, {1, 17, 2, 64}, {1, 128, 2, 64}}},
+        {PartialShape{-1, -1, -1, -1}, {{1, 1, 1, 128}, {1, 1, 1, 17}, {1, 1, 1, 128}}},
+        {PartialShape{-1, -1, -1, -1}, {{1, 128, 2, 64}, {1, 17, 2, 64}, {1, 128, 2, 64}}},
+    },
+    {
+        {PartialShape{-1, 128, -1, -1}, {{1, 128, 2, 64}}},
+        {PartialShape{-1, -1, -1, -1}, {{1, 16, 2, 64}}},
+        {PartialShape{-1, -1, 128, -1}, {{1, 1, 128, 16}}},
+        {PartialShape{-1, -1, -1, -1}, {{1, 16, 2, 32}}},
+    },
+    {
+        {PartialShape{-1, 32, -1, -1}, {{1, 32, 2, 64}}},
+        {PartialShape{-1, -1, -1, -1}, {{1, 16, 2, 64}}},
+        {PartialShape{-1, -1, 32, -1}, {{1, 1, 32, 16}}},
+        {PartialShape{-1, -1, -1, -1}, {{1, 16, 2, 32}}},
+    },
+    {
+        {PartialShape{-1, -1, -1, -1}, {{1, 16, 2, 64}}},
+        {PartialShape{-1, -1, -1, -1}, {{1, 16, 2, 64}}},
+        {PartialShape{-1, -1, 16, -1}, {{1, 1, 16, 16}}},
+        {PartialShape{-1, -1, -1, -1}, {{1, 16, 2, 32}}},
+    },
+};
+
+INSTANTIATE_TEST_SUITE_P(
+    smoke_Snippets_MHA_4D_SplitDimensionM_dynamic,
+    MHA,
+    ::testing::Combine(::testing::ValuesIn(splitm_dynamic_shapes_4d),
+                       ::testing::ValuesIn(precision_f32(4)),
+                       ::testing::Values(ov::element::f32),
+                       ::testing::Values(false),
+                       ::testing::Values(4),  // 4 Threads
+                       ::testing::Values(1),
+                       ::testing::Values(1),
+                       ::testing::Values(ov::test::utils::DEVICE_CPU),
+                       ::testing::Values(CPUTestUtils::empty_plugin_config)),
+    MHA::getTestCaseName);
+
+std::vector<std::vector<ov::test::InputShape>> splitm_dynamic_shapes_3d = {
+    {
+        {PartialShape{-1, -1, -1}, {{128, 2, 64}, {17, 2, 64}, {128, 2, 64}}},
```

Comment:
Could you please also add the test when `parallel_work_amount` is static but dimension `M` or `K` or `N` are dynamic?

Reply:
The test cases are added